From 6d727e7a41af176e390ff81d5f0c8bc629238d67 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Tue, 12 May 2015 14:54:45 +0800
Subject: [PATCH 391/462] atomisp2: Add handling for V4L2_CID_EXPOSURE_AUTO

This patch adds handling of V4L2_CID_EXPOSURE_AUTO
for atomisp2. This control is forwarded to sensor
driver.

Change-Id: Ibc205f0e73cc61aec42215d179a603b29bcaf869
Fixes-Issue: #JIMINAN-35982
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/55553
Build: IMIN (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index be0c5ba..0ec893f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -2206,6 +2206,7 @@ static int atomisp_g_ctrl(struct file *file, void *fh,
 	case V4L2_CID_2A_STATUS:
 	case V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE:
 	case V4L2_CID_EXPOSURE:
+	case V4L2_CID_EXPOSURE_AUTO:
 	case V4L2_CID_SCENE_MODE:
 	case V4L2_CID_ISO_SENSITIVITY:
 	case V4L2_CID_ISO_SENSITIVITY_AUTO:
@@ -2290,6 +2291,7 @@ static int atomisp_s_ctrl(struct file *file, void *fh,
 	switch (control->id) {
 	case V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE:
 	case V4L2_CID_EXPOSURE:
+	case V4L2_CID_EXPOSURE_AUTO:
 	case V4L2_CID_EXPOSURE_AUTO_PRIORITY:
 	case V4L2_CID_SCENE_MODE:
 	case V4L2_CID_ISO_SENSITIVITY:
@@ -2395,6 +2397,7 @@ static int atomisp_camera_g_ext_ctrls(struct file *file, void *fh,
 		ctrl.value = c->controls[i].value;
 		switch (ctrl.id) {
 		case V4L2_CID_EXPOSURE_ABSOLUTE:
+		case V4L2_CID_EXPOSURE_AUTO:
 		case V4L2_CID_IRIS_ABSOLUTE:
 		case V4L2_CID_FNUMBER_ABSOLUTE:
 		case V4L2_CID_BIN_FACTOR_HORZ:
@@ -2504,6 +2507,7 @@ static int atomisp_camera_s_ext_ctrls(struct file *file, void *fh,
 		ctrl.value = c->controls[i].value;
 		switch (ctrl.id) {
 		case V4L2_CID_EXPOSURE_ABSOLUTE:
+		case V4L2_CID_EXPOSURE_AUTO:
 		case V4L2_CID_EXPOSURE_METERING:
 		case V4L2_CID_IRIS_ABSOLUTE:
 		case V4L2_CID_FNUMBER_ABSOLUTE:
-- 
1.9.1

